docs/code/udp-dhcp/main.c

```diff
@@ -53,7 +53,11 @@ uv_buf_t make_discover_msg() {
     // HOPS
     buffer.base[3] = 0x0;
     // XID 4 bytes
+#ifdef _MSC_VER /* MSVC */
+    buffer.base[4] = (unsigned int) rand();
+#else
     buffer.base[4] = (unsigned int) random();
+#endif
```

Comment:
I think this code has a bug in that the xid is 4 bytes but this code only randomizes the first byte. Instead of rand/random, consider using uv_random: 
```suggestion
    if (uv_random(NULL, NULL, &buffer.base[4], 4, 0, NULL))
      abort();
```